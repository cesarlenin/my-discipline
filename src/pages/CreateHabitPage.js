import React, { Component } from 'react'
import { Link} from 'react-router-dom'

export class CreateHabitPage extends Component {
  render() {
    return (
    <main>
      <header >
        <h1>Create Habit</h1>
      </header>
        <section>
        <form className='signup-form'>
            <div>
              <label htmlFor="name">name</label>
              <input placeholder='read a book' type="text" name='ame' id='name' />
            </div>
            <div>
              <label htmlFor="description">description</label>
              <input type="text" name='description' id='description' placeholder='description' />
            </div>
            <div>
              <label htmlFor="Goal">Goal</label>
              <input type="number" name='Goal' id='Goal' min="1" max="7"/>
            </div>
            <Link to='/Home'>
            Submit
            </Link>
        </form>
      </section>
      </main>
    )
  }
}

export default CreateHabitPage