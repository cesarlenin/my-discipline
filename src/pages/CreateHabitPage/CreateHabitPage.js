import React, {Component} from 'react';
import HabitForm from '../../components/HabitForm/HabitForm';
import BackButton from '../../components/BackButton/BackButton';
import './CreateHabitPage.css'
export class CreateHabitPage extends Component {

  render() {
    return ( 
      <section>
        <header>
          <h1> Create Habit </h1> 
        </header> 
        <BackButton {...this.props}/> 
        <section className = 'createHabit' >
          <HabitForm {...this.props}/> 
        </section> 
      </section>
    )
  }
}

export default CreateHabitPage;