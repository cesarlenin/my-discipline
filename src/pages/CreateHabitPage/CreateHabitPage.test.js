import React from 'react';
import ReactDOM from 'react-dom';
import CreateHabitPage from './CreateHabitPage';
import renderer from 'react-test-renderer';

describe('<CreateHabitPage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <CreateHabitPage/> , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders this UI as expected', () => {
    const tree = renderer.create( <CreateHabitPage/> ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});