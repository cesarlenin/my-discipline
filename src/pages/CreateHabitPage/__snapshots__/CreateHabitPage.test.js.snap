// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateHabitPage /> renders this UI as expected 1`] = `
<section>
  <header>
    <h1>
       Create Habit 
    </h1>
  </header>
  <div
    className="back"
  >
    <button
      onClick={[Function]}
    >
      Go Back
    </button>
  </div>
  <section
    className="createHabit"
  >
    <form
      className="habit-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="name"
      >
         Name: 
      </label>
      <input
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="read a book"
        type="text"
      />
      <label
        htmlFor="description"
      >
         Description: 
      </label>
      <input
        id="description"
        name="description"
        onChange={[Function]}
        placeholder="description"
        type="text"
      />
      <label
        htmlFor="Goal"
      >
         Set a Goal, how many times a week ? 
      </label>
       
      <input
        id="Goal"
        max="7"
        min="1"
        name="Goal"
        onChange={[Function]}
        type="number"
      />
      <button
        disabled="Name is required"
        type="submit"
        value="Submit"
      >
        Submit
      </button>
    </form>
  </section>
</section>
`;
